static/app/components/events/featureFlags/eventFeatureFlagSection.tsx

```diff
@@ -121,21 +129,47 @@ function BaseEventFeatureFlagList({event, group, project}: EventFeatureFlagSecti
   );
 
   const {
-    suspectFlags,
-    isError: isSuspectError,
-    isPending: isSuspectPending,
+    suspectFlags: legacySuspectFlags,
+    isError: isLegacySuspectError,
+    isPending: isLegacySuspectPending,
   } = useLegacyEventSuspectFlags({
+    enabled: !enableSuspectFlags, // Fallback to the legacy strategy
     organization,
     firstSeen: group.firstSeen,
     rawFlagData,
     event,
   });
 
+  const {
+    data: suspectScores,
+    isError: isSuspectError,
+    isPending: isSuspectPending,
+  } = useGroupSuspectFlagScores({
+    groupId: group.id,
+    environment: environments.length ? environments : undefined,
+    enabled: enableSuspectFlags,
+  });
+
   const suspectFlagNames: Set<string> = useMemo(() => {
-    return isSuspectError || isSuspectPending
+    if (enableSuspectFlags) {
+      return isSuspectError || isSuspectPending
+        ? new Set()
+        : new Set(
+            suspectScores.data.filter(score => score.score >= 7).map(score => score.flag)
+          );
+    }
+    return isLegacySuspectError || isLegacySuspectPending
       ? new Set()
-      : new Set(suspectFlags.map(f => f.flag));
-  }, [isSuspectError, isSuspectPending, suspectFlags]);
+      : new Set(legacySuspectFlags.map(f => f.flag));
+  }, [
```

Comment:
are the order of ops right here?

Reply:
like the `||` vs the ternary? yes. the linter is removing round braces when i put them in. 

it's not obvious though unless ppl know the operator precedence. i'll see if i can write it in a more obvious way

Reply:
optional chaining! we're returning empty an `Set` in either if the `map` call has nothing in it.